hivemind/averaging/allreduce.py

```diff
@@ -132,6 +148,8 @@ def should_delay_results(self, peer_id: PeerID) -> bool:
     async def run(self) -> AsyncIterator[torch.Tensor]:
         """Run all-reduce, return differences between averaged and original tensors as they are computed"""
         pending_tasks = set()
+        if self.sender_timeout is not None:
+            asyncio.create_task(self._handle_missing_senders())
```

Comment:
Such fire-and-forget calls always lead to "Task destroyed but is pending". Please fix that, e.g., save the task to a class field and `await` when finalizing.

Reply:
GREAT catch: i've meant to add it to pending_tasks (and i just did so)